test-clients/src/main/java/com/hedera/services/bdd/spec/queries/contract/HapiGetContractRecords.java

```diff
@@ -206,8 +206,13 @@ private void checkExpectations(HapiApiSpec spec, List<TransactionRecord> records
 				Assert.assertEquals("Wrong record #" + i, expected, records.get(i));
 			}
 		} catch (Exception e) {
-			log.error("Something amiss with the expected records...", e);
-			Assert.fail("Impossible to meet expectations (on records)!");
+			if(log.isDebugEnabled()) {
+				log.error("Something amiss with the expected records...", e);
+			}
+			else {
+				log.error("Something amiss with the expected records {}", records);
+			}
+			throw new Exception("Impossible to meet expectations (on records)!");
```

Comment:
why not throw a newly created query exception?

Reply:
Yeah, fixed.